### DISCUSSION TITLE
How to set object relations via Netbox Shell?

### DISCUSSION THREAD

**Discussion by LukasIgel** [netbox-community/netbox#11684] [Q&A] [answered]
Hello everyone,

I am trying to programmatically fill my NetBox instance with data via NetBox shell. I am struggling with setting object relations, for example: 

Creating a tenant group object, and afterwards creating a new tenant object and setting it's tenant group to the mentioned tenant group. 

Maybe you can search all instances of "TenantGroup" for the needed object, but I don't know how to assign it's reference to the tenant object. 

Thanks for your help! -Igel

**Comment by kkthxbye-code** [comment, marked as answer]
```
>>> tg = TenantGroup(name="testgroup", slug="testgroup")
>>> tg.full_clean()
>>> tg.save()
>>> t = Tenant(name="testtenant", slug="testtenant", group=tg)
>>> t.full_clean()
>>> t.save()
```

However I do not recommend using nbshell to import data as change logging does not work by default in it.

**Reply by LukasIgel** [reply]
Thanks, would you recommend using the REST API instead?

**Reply by kkthxbye-code** [reply]
The REST API is a good option, if that's too slow another option is to use a custom script. Lastly JSON/YAML/CSV import is also a popular option.

If you decide on using either the shell or a custom script, remember to always call full_clean() before saving to make sure validation is executed.
